kuma/wiki/views/list.py

```diff
@@ -190,30 +189,32 @@ def get_previous(revisions):
 
     # Grab revisions, but defer summary and content because they can lead to
     # attempts to cache more than memcached allows.
-    revisions = MultiQuerySet(
-        (Revision.objects.filter(pk=document.current_revision.pk)
-                         .prefetch_related('creator', 'document')
-                         .transform(get_previous)),
-        (Revision.objects.filter(document=document)
-                         .order_by('-created', '-id')
-                         .exclude(pk=document.current_revision.pk)
-                         .prefetch_related('creator', 'document')
-                         .transform(get_previous))
-    )
-
-    if not revisions.exists():
+    all_revisions = (document.revisions.defer('summary', 'content').order_by('created', 'id')
```

Comment:
I enabled Django Debug Toolbar locally to investigate.  This new version takes 22 queries in 19.23 ms, and the old version takes 17 queries in 10.56ms, on Mozilla/Firefox, which has 15 revisions in the sample database.  I believe the extra requests is because the ``creator`` is pre-fetched in the previous version, and in the new one it requires an additional database query.  I think a ``prefetch_related('creator')`` is needed, to better compare the two. I think ``prefetch_related('document')`` can be skipped, but I might be guessing at Django database query implementation.